SpriteSheetPacker/PublishSpriteSheet.cpp

```diff
@@ -39,7 +39,7 @@ void PublishSpriteSheet::addSpriteSheet(const SpriteAtlas &atlas, const QString
     _fileNames.append(fileName);
 }
 
-bool PublishSpriteSheet::publish(const QString& format, const QString& optMode, int optLevel, bool errorMessage) {
+bool PublishSpriteSheet::publish(const QString& format, const QString& optMode, int optLevel, bool errorMessage, PublishStatusDialog* dialog) {
```

Comment:
call calback and remove PublishStatusDialog* dialog
plz void (*Callback)(int currentStep, int allSteps)

and call cb(0, n) cb(1, n) .....

publish(......, [](int a, int b)) {
  publishStatusDialog->progresBar(...)
}